### PR TITLE
Posts about Patterns

### PR DESCRIPTION
**Pull request by pmn**
Creates two posts, one announcing Patterns, and one about how CascadiaJS is using them. 

@haacked mind improving the announcing Patterns post?

### CODE REVIEW

File: _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Test the skill in the console by entering the domain of a website you want to check. In the console, you will need to include `https://` in front of the domain name. 
+
+Once you’ve verified that the skill works, add a Pattern by clicking the link that says “Manage skill patterns” on the left side of the screen. Click “Create Pattern”, and give the new pattern a descriptive name like “url pattern”. To save time, pick a pattern type of “Regular Expression” and use this regular expression: 
+
+`is (<)?(http:\/\/|https:\/\/)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)(|.*?)(>?) down`
+
+This Pattern is made a little more complicated by all the ways chat platforms might send us the URL, for now you'll just have to trust that it works. You can see for yourself by using the built-in tester. Try a phrase like “is google.com down?” to make sure it matches.
+
+Once this is enabled, jump into any chat room that has Abbot in it, and try asking if a site is up or down. Go on, we’ll wait for you to try it!
```
Comment (shana) [inline _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md:50]: Great example, love it!

**Review by shana — APPROVED** [review]

File: _posts/2021/2021-11-03-introducing-patterns.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+author:
+    avatar: https://2.gravatar.com/avatar/cdf546b601bf29a7eb4ca777544d11cd?s=160
+    name: haacked
+---
+
+Until recently, Abbot skills could only be called by calling them directly by name (i.e. “@abbot grafana help”). This makes Abbot easy to get started with and reason about. However, there are entire categories of problems that Abbot can help out with where it doesn’t make sense to call a command. 
+
+Since Abbot can respond to ambient information, we’re excited to announce a brand new feature called Patterns. Patterns allow you to configure listeners in Abbot that pay attention to specific words and phrases. Patterns can be configured to look at the beginning or end of a sentence, or even a regular expression. 
+
+We think that Patterns will be useful to your team, because they can connect Abbot skills to everyday conversations that your team has in chat. There are simple use cases like keeping track of karma points or checking if a website is down, or more complex ones like linking to external issues, or even monitoring chat for errant yubikey clicks.
```
Comment (pmn) [inline _posts/2021/2021-11-03-introducing-patterns.md:15]: Karma package: https://ab.bot/packages/aseriousbiz/karma

File: _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+title: "How CascadiaJS uses Abbot to power its live feed"
+description: "CascadiaJS uses Abbot patterns to power its live feed. See how to use Patterns in this post"
+tags: [patterns,cascadiajs,abbot]
+excerpt_image: 
+author:
+    avatar: https://avatars.githubusercontent.com/u/279389?s=400&u=74e6e598a2bf9d9889fbb771c542c508bf270e36&v=4
+    name: pmn
+---
+
+Attendees to CascadiaJS saw updates from the Cascadia chat from the Cascadia Discord as soon as the messages were sent.
```
Comment (haacked) [inline _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md:11]: This sentence is a bit awkward and hard to parse. Where did they see these updates? Maybe,

```suggestion
During the CascadiaJS conference, attendees could see live updates on screens around the conference with the latest chat messages from the Cascadia Discord. 
```

Or something like that?

File: _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+Since this was done for a JavaScript conference, let's build our script using JavaScript. Log in to Abbot and create a new JavaScript skill called `website-checker`. Copy and paste the code below into your skill:
+
+```javascript
+  const axios = require('axios');
+
+  let website = bot.arguments; // bot.arguments contains the user-provided text
+  if (bot.isPatternMatch) {    // this will be true if it's the result of a pattern match. Otherwise, it's a chat message.
+    website = bot.tokenizedArguments[1].toString(); // Abbot tokenizes arguments to make it easier to ahandle them.
+  }
```
Comment (haacked) [inline _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md:29]: This is a stylistic choice, but if this were my code, I'd probably do this. :smile:

```suggestion
  // bot.arguments contains the user-provided text
  // bot.tokenizedArguments contains parsed arguments to make it easier to handle them.
  const website = bot.isPatternMatch ? bot.tokenizedArguments[1].value : bot.arguments;
```

File: _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  
+  await axios.get(website)
+    .then(function (response) {
+        // handle success
+        bot.reply(`${bot.from} it's just you.. it looks okay to me...`)
```
Comment (haacked) [inline _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md:34]:
```suggestion
        await bot.reply(`${bot.from} it's just you.. it looks okay to me...`)
```

File: _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      })
+      .catch(function (error) {
+        // handle error
+        bot.reply(`it's not just you ${bot.from}, i got this error when i tried: \n ${error}`)
```
Comment (haacked) [inline _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md:38]:
```suggestion
        await bot.reply(`it's not just you ${bot.from}, i got this error when i tried: \n ${error}`)
```

**Review by haacked — COMMENTED** [review]
A few minor suggestionts.

File: _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+title: "Updating the CascadiaJS live stream with Abbot"
+description: "CascadiaJS uses Abbot patterns to power its live feed. See how to use Patterns in this post"
+tags: [patterns,cascadiajs,abbot]
+excerpt_image: 
+author:
+    avatar: https://avatars.githubusercontent.com/u/279389?s=400&u=74e6e598a2bf9d9889fbb771c542c508bf270e36&v=4
+    name: pmn
+---
+
+If you attended CascadiaJS 2021, you may have noticed live updates on the website that came from Cascadia's Discord server.
```
[the remark this answers is not in the document]
Comment (pmn) [reply _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md:11]: Agree, but I haven't been able to access it yet. May grab one tomorrow once it's live.

File: _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md (hunk 5 above)
Comment (pmn) [reply _posts/2021/2021-11-03-how-cascadiajs-uses-abbot.md:33]: For whatever reason, this doesn't work inside of the `await`ed `axios.get` (it complains about not being inside of an `async` function). Going to leave it as is because it's functioning correctly on test.
